notification-server/src/bin/service.rs

```diff
@@ -112,35 +112,59 @@ async fn traverse_chain(
                 continue;
             }
         };
-        for result in process(transactions).await.iter() {
+        for result in process(transactions).await.into_iter() {
             info!(
                 "Sending notification to account {} with type {:?}",
-                result.address, result.transaction_type
+                result.address(),
+                result.transaction_type()
             );
-
             let devices = loop {
                 match database_connection
                     .prepared
-                    .get_devices_from_account(result.address)
+                    .get_devices_from_account(result.address())
                     .await
                 {
                     Ok(devices) => break devices,
                     Err(err) => {
                         error!(
                             "Error retrieving devices for account {}: {:?}. Retrying...",
-                            result.address, err
+                            result.address(),
+                            err
                         );
                         sleep(DATABASE_RETRY_DELAY).await;
                     }
                 }
             };
-
-            for device in devices
+            let devices: Vec<_> = devices
                 .iter()
-                .filter(|device| device.preferences.contains(&result.transaction_type))
-            {
+                .filter(|device| device.preferences.contains(result.transaction_type()))
+                .collect();
```

Comment:
This collect should not be required.

If someone has a good solution (For example using peekable), then that is fine. 

Else this is some pseudo optimizations which will not matter in practice.